Sources/NIOCore/ByteBuffer-hex.swift

```diff
@@ -263,3 +283,23 @@ extension ByteBuffer {
         }
     }
 }
+
+extension String {
+    /// Plain decode a string representing a hexadecimal sequence them into an UInt8 sequence
+    /// - Complexity: O(n)
+    public var hexPlainDecodedBytes: [UInt8] {
```

Comment:
This should not be `public`.

We also don't want to round-trip it through an Array: we should be able to convert directly into `ByteBuffer`.

Reply:
![Screenshot 2024-08-13 at 11 39 23 am](https://github.com/user-attachments/assets/91c512a5-7855-44c9-bc2f-75a08cf2c7c1)

Without turning it into an array we lose 2 things 

1. The ability to count the array for the exact number of bytes. The underestimated count can give 0 for sequence length that that are really 11 and it then means either I cannot allocate a buffer of the correct length ( or return early without setting any bytes). 

So if in the initliser someone tried to enter a hex string without anything in it, it would still return an empty buffer, but after some extrannuous computation and going through the setBytes slow path.

Or again, if someone tried to write empty bytes.

2. It always uses the slow method for setting bytes in the sequence rather than the contiuous if available which is done always in an array from my testing. 

Is this extra rount trip from sequence to array worth more than the cost of the above? It is becoming a difficult question for me to answer. After your comment I am leaning towards it is not worth the extra cost, seems like there is a change it takes the slow path ANYWAYS with an array. So the fast case might be faster, but the slow case slower. 

Putting my brain dump here for challenging the outcome or the thought process.

Reply:
Made the changes. Take a look when you find the time